<?xml version="1.0"?>
<launch>



<!-- Edit this part optionally for debugging-->
  <arg name="gui" default="true"/> <!-- Gazebo gui, set false when too slow -->
  <arg name="pause" default="false"/> <!-- Gazebo start with paused -->
  <arg name="verbose" default="false"/> <!-- Gazebo verbose, set true when debug Gazebo -->

<!-- Edit this part with your own algorithms -->
  <node name="example_control_node" pkg="khnp_competition" type="example_control_code.py"/>





<!-- Do not touch below!!!!!!!! -->
 <!-- course changing is only allowed for competition managers -->
  <arg name="course" default="practice"/> <!-- practice: open source for participants, real: not open -->

  <!-- Gazebo -->
  <arg if="$(eval arg('course') == 'practice')" name="world" value="$(find khnp_competition)/../gazebo_map_for_khnp/practice.world"/>
  <arg if="$(eval arg('course') == 'real')" name="world" value="$(find khnp_competition)/../gazebo_map_for_khnp/real.world" />
  <include file="$(find khnp_competition)/launch/gazebo.launch">
      <arg name="world" value="$(arg world)"/>
      <!-- <arg name="world" value="worlds/empty.world"/> -->
      <arg name="gui" value="$(arg gui)" />
      <arg name="verbose" value="$(arg verbose)"/>
      <arg name="pause" value="$(arg pause)"/>
  </include>

  <!--  MAIN code -->
  <param name="/img_width" value="480"/> <!-- only for visualization, not for real image width -->
  <param name="/img_height" value="360"/> 
  <param name="/first_cam_topic" value="/l515/depth/rgb_image_raw/compressed"/>
  <param name="/third_cam_topic" value="/third_camera/rgb/image_raw/compressed"/>
  <param name="/third_cam_name" value="third_camera"/>
  <param name="/robot_name" value="iris"/>

  <node name="khnp_competition_code" pkg="khnp_competition" type="khnp_competition_node" output="screen"/>
  <!-- launch-prefix="gdb -ex run --> <!--args" -->
  <node name="khnp_competition_spawner" pkg="khnp_competition" type="spawning.py" output="screen"/>
  <node name="khnp_competition_tf_broadcaster" pkg="khnp_competition" type="tf_broadcaster.py" output="screen"/>

</launch>
